base_layer/core/src/base_node/state_machine_service/states/horizon_state_sync/horizon_state_synchronization.rs

```diff
@@ -240,7 +255,12 @@ impl<'a, B: BlockchainBackend + 'static> HorizonStateSynchronization<'a, B> {
         Ok(())
     }
 
-    async fn synchronize_outputs(&mut self, to_header: &BlockHeader) -> Result<(), HorizonSyncError> {
+    async fn synchronize_outputs(
```

Comment:
The function was previously split in two just for the sake of making it somewhat shorter without encapsulating anything (it was called `sync_output_nodes` but may as well be called `synchronize_outputs_inner`). Nor did it improving readability, reduce code duplication or reduce cognitive load. I did not want to pass the client from the outer to the inner (I'm sure you'd agree, passing additional bytes over the wire to avoid adding a single function parameter is an anti-pattern).